flask_monitor/metrics.py

```diff
@@ -77,29 +89,95 @@ def after_request(response):
+            "http", response.status_code, error_status, "", request.method, request.path
         ).inc(size_request)
         return response
+
     if error_fn is not None:
-        is_error.__code__ = error_fn.__code__
+        _is_error_.__code__ = error_fn.__code__
     app.before_request(before_request)
     app.after_request(after_request)
+    return app, registry
+
 
-def watch_dependencies(dependency, func, time_execution=1500):
+def watch_dependencies(dependency, func, time_execution=15000, registry=None, app=current_app):
+    """
+    Register dependencies metrics up
+    """
+
+    if not registry:
+        registry = app.extensions.get("registry", CollectorRegistry())
+        app.extensions["registry"] = registry
+
+    # pylint: disable=invalid-name
+    DEPENDENCY_UP = Gauge(
+        'dependency_up',
+        'records if a dependency is up or down. 1 for up, 0 for down',
+        ["name"],
+        registry=registry
+    )
+    def register_dependecy():
+        DEPENDENCY_UP.labels(dependency).set(func())
+
+    scheduler = BackgroundScheduler()
+    scheduler.add_job(
+        func=register_dependecy,
+        trigger="interval",
+        seconds=time_execution/1000,
+        max_instances=1,
+        name='dependency',
+        misfire_grace_time=2,
+        replace_existing=True
+        )
+    scheduler.start()
+
+    # Shut down the scheduler when exiting the app
+    atexit.register(scheduler.shutdown)
+    return scheduler
+
+# pylint: disable=too-many-arguments
+def collect_dependency_time(
+    app, name, rtype='http', status=200,
+    is_error=False, error_message='',
+    method='GET', addr='/', **kwargs):
     """
     Register dependencies metrics
     """
-    def thread_function():
-        thread = threading.Timer(time_execution, lambda x: x + 1, args=(1,))
-        thread.start()
-        thread.join()
-        response = func()
-        DEPENDENCY_UP.labels(dependency).set(response)
-        thread_function()
-    thread = threading.Timer(time_execution, thread_function)
-    thread.start()
+
+    # pylint: disable=global-statement
+    global DEPENDENCY_UP_LATENCY
+
+    if not kwargs.get('registry'):
+        registry = app.extensions.get("registry", CollectorRegistry())
+        app.extensions["registry"] = registry
+
+    if not DEPENDENCY_UP_LATENCY:
+        DEPENDENCY_UP_LATENCY = app.extensions.get(
+            "DEPENDENCY_UP_LATENCY",
+            Histogram(
+                "dependency_request_seconds",
+                "records in a histogram the number of requests to dependency",
+                ["name", "type", "status", "isError", "errorMessage", "method", "addr"],
+                registry=registry
+            )
+        )
+        app.extensions['DEPENDENCY_UP_LATENCY'] = DEPENDENCY_UP_LATENCY
+
+    elapsed = kwargs.get('elapsed')
+    if not elapsed:
+        elapsed = time.time() - kwargs.get('start', time.time())
```

Comment:
@Ziul, this logic is weird. Why don't we adopt something more like express-monitor's `collectDependencyTime` func? https://github.com/labbsr0x/express-monitor/blob/master/lib/monitor.ts

Reply:
If the user passes the elapsed time, like from the response object in the example, is more precise than we calculate the time inside our method. But he can pass the start time as well, both are acceptable, but elapsed is more accurate.

Reply:
@Ziul, it's more logic delegated to the user. Nonetheless, you are getting the elapsed time from `kwargs`. Why not more explicit?

Reply:
Code updated without `kwargs`.